apps/beeai-server/src/beeai_server/api/auth.py

```diff
@@ -166,7 +166,7 @@ async def introspect_token(token: str, configuration: Configuration) -> tuple[di
 async def decode_oauth_jwt_or_introspect(
     configuration: Configuration, token: str, jwks_dict: JwksDict | None = None, aud: str | None = None
 ) -> tuple[dict | None, str | None]:
-    if jwks_dict:
+    if jwks_dict and token:
```

Comment:
[nitpick] if `token` is `None` it would be better to not even call the decode function